kivy/uix/textinput.py

```diff
@@ -835,7 +856,186 @@ def _set_undo_redo_bkspc(self, ol_index, new_index, substring, from_undo):
         #reset redo when undo is appended to
         self._redo = []
 
-    def do_cursor_movement(self, action):
+    def _move_word_left(self, index=None):
```

Comment:
In fact I thunk it should be possible to combine these two functions on to one `_move_cursor(movement_mode='word-left')`?

Where movement_mode can be one of `word-left` `word-right`, `line-up`, `line-down`, 'home', `end`?